internal/statemachine/common_states.go

```diff
@@ -105,81 +104,76 @@ func (stateMachine *StateMachine) generateDiskInfo() error {
 
 var calculateRootfsSizeState = stateFunc{"calculate_rootfs_size", (*StateMachine).calculateRootfsSize}
 
-// Calculate the size of the root filesystem
-// on a 100MiB filesystem, ext4 takes a little over 7MiB for the
-// metadata. Use 8MB as a minimum padding here
+// calculateRootfsSize calculates the size of the root filesystem.
+// On a 100MiB filesystem, ext4 takes a little over 7MiB for the
+// metadata, so use 8MB as a minimum padding.
 func (stateMachine *StateMachine) calculateRootfsSize() error {
-	// use `du` to calculate the size of the rootfs
 	rootfsSize, err := helper.Du(stateMachine.tempDirs.rootfs)
```

Comment:
I know this code was already here but anyway, if the underlying filesystem uses transparent compression this method will return the wrong size (unless rootfs is guaranteed to be mounted on an FS that doesn't use compression). For example, on ZFS `du` will return the space the file is occupying on disk, not its real size.

Reply:
Good catch, I was not aware of that. No, there is no guarantee ubuntu-image will run on a FS that doesn't use compression, to this indeed might return a wrong size. This may be the source of bugs some of our users encounter. I will open a bug to fix that because I am not sure the possible quick fix I see could be more reliable than the current one.